packages/admin-api-client/src/tests/admin-api-client.test.ts

```diff
@@ -0,0 +1,356 @@
+          getApiUrl: expect.any(Function),
+          request: expect.any(Function),
+          fetch: expect.any(Function),
+        });
+      });
+
+      describe("validations", () => {
+        it("throws an error when a store domain is not provided", () => {
+          expect(() =>
+            createAdminApiClient({
+              ...config,
+              storeDomain: undefined as any,
+            })
+          ).toThrow(
+            new Error(
+              'Admin API Client: a valid store domain ("undefined") must be provided'
+            )
+          );
+        });
+
+        it("throws an error when an empty string is provided as the store domain", () => {
+          expect(() =>
+            createAdminApiClient({
+              ...config,
+              storeDomain: "   " as any,
+            })
+          ).toThrow(
+            new Error(
+              'Admin API Client: a valid store domain ("   ") must be provided'
+            )
+          );
+        });
+
+        it("throws an error when the provided store domain is not a string", () => {
+          expect(() =>
+            createAdminApiClient({
+              ...config,
+              storeDomain: 123 as any,
+            })
+          ).toThrow(
+            new Error(
+              'Admin API Client: a valid store domain ("123") must be provided'
+            )
+          );
+        });
+
+        it("throws an error when the api version is not provided", () => {
+          expect(() =>
+            createAdminApiClient({
+              ...config,
+              apiVersion: undefined as any,
+            })
+          ).toThrow(
+            new Error(
+              `Admin API Client: the provided apiVersion ("undefined") is invalid. Current supported API versions: ${mockApiVersions.join(
+                ", "
+              )}`
+            )
+          );
+        });
+
+        it("console warns when a unsupported api version is provided", () => {
+          const consoleWarnSpy = jest
+            .spyOn(global.console, "warn")
+            .mockImplementation(jest.fn());
+
+          createAdminApiClient({
+            ...config,
+            apiVersion: "2022-07",
+          });
+
+          expect(consoleWarnSpy).toHaveBeenCalledWith(
+            `Admin API Client: the provided apiVersion ("2022-07") is deprecated or not supported. Current supported API versions: ${mockApiVersions.join(
+              ", "
+            )}`
+          );
+        });
+
+        it("throws an error when an access token isn't provided", () => {
+          expect(() =>
+            createAdminApiClient({
+              ...config,
+              accessToken: undefined as any,
+            })
+          ).toThrow(
+            new Error(`Admin API Client: an access token must be provided`)
+          );
+        });
+
+        it("throws an error when run in a browser environment (window is defined)", () => {
+          global.window = {} as any;
```

Comment:
Wouldn't this still be handy to catch someone removing the browser/`window` check?

Reply:
The test itself is fine - it's the `global.window` code I think we can remove. 

Since these tests run only in a server environment, `global.window` should already be undefined.